### DISCUSSION TITLE
Randomly populating Milky Way Satellite Galaxies

### DISCUSSION THREAD

**Discussion by soumyodiptak** [galacticusorg/galacticus#586] [Q&A]
Dear Andrew,

I have just started to understand the Galacticus software. I am trying from scratch to populate MW satellites randomly by using Galacticus for LambdaCDM model. I am getting errors. I have shared my code and the error below. Can you please check and guide me? I tried to make the code very simple so that I can understand it first.

<?xml version="1.0" encoding="UTF-8"?>

<galacticus>
    
    <cosmology>
        <HubbleParameter value="0.7"/> 
        <OmegaMatter value="0.3"/> 
        <OmegaBaryon value="0.04"/> 
        <OmegaLambda value="0.7"/> 
        <sigma8 value="0.8"/> 
        <ns value="0.96"/> 
    </cosmology>

    
    <parameters>
        <model type="diskInstability">
            <parameter name="diskInstabilityToft2002Frac" value="0.05"/> 
        </model>
    </parameters>

    
    <galaxies>
        <galaxy>
            <position value="0.0, 0.0, 0.0"/> 
            <mass value="1.0e12"/> 
        </galaxy>
    </galaxies>

    
    <satellites>
        <model type="subhaloAbundanceMatching">
            <parameter name="stellarMassFunction" value="Baldry2012"/> 
        </model>
    </satellites>

    
    <satellitePopulation>
        <randomPopulation>
            <numberOfSatellites value="100"/> 
        </randomPopulation>
    </satellitePopulation>

    
    <satelliteOrbits>
        
        <orbit>
            <mass value="1.0e10"/> 
            <radius value="200.0"/> 
        </orbit>
        
        <orbit>
            <mass value="5.0e9"/> 
            <radius value="250.0"/> 
        </orbit>
    </satelliteOrbits>

    
    <darkMatter>
        <haloMassFunction>
            <model type="Tinker2008"/> 
            <file>./static/darkMatter/Halo_Mass_Function_Parameters_Tinker_2008.xml</file>     
        </haloMassFunction>
    </darkMatter>

    
    <darkMatter>
        <haloDensityProfile>
            <model type="NFW"/> 
        </haloDensityProfile>
    </darkMatter>

    
    <galaxyHaloConnection>
        <model type="Behroozi2013"/> 
    </galaxyHaloConnection>

    
    <tidalEffects>
         <model type="satelliteTidalEffects"/> 
    </tidalEffects>

    
    <simulation>
        <boxSize value="50.0"/> 
        <outputRedshifts>
            <outputRedshift value="0.0"/> 
            <outputRedshift value="1.0"/>
        </outputRedshifts>
        <outputFormat>
            <outputFormat type="HDF5"/> 
        </outputFormat>
        <outputPrefix value="MW_Satellite_Galaxies_"/> 
    </simulation>

</galacticus>

I am getting the following error:

Fatal error:
orbit masses and radius must be specified
 Occurred at:
     function:Kepler_Orbits_Velocity_Scale
       module:Kepler_Orbits
         file:objects.kepler_orbits.F90   [line 920]

Best regards,
Soumyodipta Karmakar

**Comment by abensonca** [comment]
Hi Soumyodipta. Your parameter file doesn't follow the syntax that Galacticus expects. All parameters in the file will be of the form:
```
<name value="xyz"/>
```
where `name` is the parameter name, and `xyz` is the value that the parameter should take.

Also, many of your parameters are not things that are defined in Galacticus. For example, `galaxyHaloConnection` simply isn't a thing that exists in Galacticus, and constructions like:
```
<darkMatter>
    <haloDensityProfile>
        <model type="NFW"/> <!-- Dark matter density profile model -->
    </haloDensityProfile>
</darkMatter>
```
will not work because both `darkMatter` and `haloDensityProfile` are not entities that are defined in Galacticus.

Can you explain how you created this parameter file? Were you working from an example? Currently, it's just not in a form that Galacticus can understand so I'm not able to give much guidance on what you need to do to build a working model - I would first need to understand what it is that you're trying to achieve.

**Comment by soumyodiptak** [comment]
Dear Andrew,

Since I am a beginner on this, I am trying to understand by trial and error. My goal is to write a code to simulate MW satellite population along with their features based on LambdaCDM model.

**Comment by abensonca** [comment]
Hi Soumyodipta. I'd recommend starting from an existing example in that case - setting up a complete model for galaxy formation is very non-trivial. Here's an example that you could start with:
```
<?xml version="1.0" encoding="UTF-8"?>
<!-- Parameters of a Milky Way model. -->
<parameters>
  <formatVersion>2</formatVersion>
  <version>0.9.4</version>

  <!-- Include required parameters -->
  <xi:include href="parameters/reference/darkMatterParticleCDM.xml"       xpointer="xpointer(parameters/*)" xmlns:xi="http://www.w3.org/2001/XInclude"/>
  <xi:include href="parameters/reference/cosmology.xml"                   xpointer="xpointer(parameters/*)" xmlns:xi="http://www.w3.org/2001/XInclude"/>
  <xi:include href="parameters/reference/powerSpectrum.xml"               xpointer="xpointer(parameters/*)" xmlns:xi="http://www.w3.org/2001/XInclude"/>
  <xi:include href="parameters/reference/structureFormation.xml"          xpointer="xpointer(parameters/*)" xmlns:xi="http://www.w3.org/2001/XInclude"/>
  <xi:include href="parameters/reference/darkMatterHalosProfile.xml"      xpointer="xpointer(parameters/*)" xmlns:xi="http://www.w3.org/2001/XInclude"/>
  <xi:include href="parameters/reference/darkMatterHalosTidalHeating.xml" xpointer="xpointer(parameters/*)" xmlns:xi="http://www.w3.org/2001/XInclude"/>
  <xi:include href="parameters/reference/darkMatterHalosStructure.xml"    xpointer="xpointer(parameters/*)" xmlns:xi="http://www.w3.org/2001/XInclude"/>
  <xi:include href="parameters/reference/subhaloOrbits.xml"               xpointer="xpointer(parameters/*)" xmlns:xi="http://www.w3.org/2001/XInclude"/>
  <xi:include href="parameters/reference/mergerTrees.xml"                 xpointer="xpointer(parameters/*)" xmlns:xi="http://www.w3.org/2001/XInclude"/>
  <xi:include href="parameters/reference/evolutionGalaxyFormation.xml"    xpointer="xpointer(parameters/*)" xmlns:xi="http://www.w3.org/2001/XInclude"/>

  <!-- Set verbosity to standard level. -->
  <verbosityLevel value="standard"/>

  <!-- Set up the tasks to perform. -->
  <task value="evolveForests">
    <!-- Evolve merger tree forests. -->
  </task>

  <!-- Merger tree building options -->
  <mergerTreeBuildMasses value="fixedMass">
    <!-- Halo masses fixed at a single value. -->
    <!-- The value used is the median estimate at r=262kpc from Slizewski et al. (2022; https://ui.adsabs.harvard.edu/abs/2022ApJ...924..131S) -->
    <massTree   value="1.520e12"/>
    <radiusTree value="0.262e00"/>
    <treeCount  value="1"       />
  </mergerTreeBuildMasses>

  <!-- Halo mass resolution -->
  <mergerTreeMassResolution value="fixed">
    <!-- All trees are set to have the same halo mass resolution. -->
    <!-- A relatively coarse mass resolution is used to allow the model to run in reasonable time. -->
    <massResolution value="8.0e7"/>
  </mergerTreeMassResolution>

  <!-- Output epochs -->
  <outputTimes value="list">
    <!-- Additional redshifts can be added to this list to generate more outputs. -->
    <redshifts value="0.0"/>
  </outputTimes>

  <!-- Output -->
  <mergerTreeOutputter value="standard"/>

  <!-- Output file -->
  <outputFileName value="milkyWay.hdf5"/>

</parameters>
```

This will run a single MW merger tree and output results at $z=0$. 

Note that it has a relatively coarse resolution of 8 x 10⁷ M☉. A resolution of 10⁷ M☉ would be better but will take longer to run.

Note that this parameter file should be placed in the main Galacticus folder as it needs to read other parameter files located in `parameters/reference/`.

This model is the one published in [Ahvazi et al. (2024)](https://ui.adsabs.harvard.edu/abs/2023arXiv230813599A/abstract).
